src/Driver.java

```diff
@@ -0,0 +1,101 @@
+import java.util.*;
+
+import Models.*;
+import Services.SnakeAndLadderGameService;
+
+public class Driver {
+    public static void main(String args[]) {
+        int noOfSnakes, noOfLadders, noOfPlayers, noOfDices, boardSize;
+        ArrayList<Snake> snakes = new ArrayList<Snake>();
+        ArrayList<Ladder> ladders = new ArrayList<Ladder>();
+        Queue<Player> players = new ArrayDeque<Player>();
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("Enter number of dices to use in game");
+        noOfDices = sc.nextInt();
+
+        System.out.println("Enter board size");
+        boardSize = sc.nextInt();
+
+        System.out.println("Enter Number of Snakes");
+        noOfSnakes = sc.nextInt();
+        System.out.println("Enter Position of head and tail of " + noOfSnakes + " Snakes");
+        for(int i = 0; i < noOfSnakes; i++) {
+            int head, tail;
+            head = sc.nextInt();
+            tail = sc.nextInt();
+            snakes.add(new Snake(head, tail));
+        }
+
+        System.out.println("Enter Number of Ladders");
+        noOfLadders = sc.nextInt();
+        System.out.println("Enter Position of start and end of " + noOfLadders + " Ladders");
+        for(int i = 0; i < noOfLadders; i++) {
+            int start, end;
+            start = sc.nextInt();
+            end = sc.nextInt();
+            ladders.add(new Ladder(start, end));
+        }
+
+        System.out.println("Enter Number of Players");
+        noOfPlayers = sc.nextInt();
+        System.out.println("Enter Names of " + noOfPlayers + " Players");
+        for(int i = 0; i < noOfPlayers; i++) {
+            System.out.println("Enter Player Name: ");
+            String name;
+            name = sc.next();
+            players.add(new Player(name));
+        }
+
+        Board board = new Board(boardSize, snakes, ladders);
+        SnakeAndLadderGameService gameService = new SnakeAndLadderGameService(noOfPlayers, noOfDices, board, players);
+
+        System.out.println("Do you want to end game after getting first winner: Yes/No");
+        while(true) {
+            String endGameAfterFirstWinner;
+            endGameAfterFirstWinner = sc.nextLine();
+            if(endGameAfterFirstWinner.equalsIgnoreCase("YES")) {
+                gameService.setEndGameAfterFirstWinner(true);
+                break;
+            }
+            else if(endGameAfterFirstWinner.equalsIgnoreCase("NO")) {
+                gameService.setEndGameAfterFirstWinner(false);
+                break;
+            }
+            else {
+                System.out.println("Please Enter 'Yes' or 'No'");
+            }
+        }
+
+        Queue<Player> leaderBoard = gameService.startGame();
```

Comment:
This logic is core to the game and should not be in Driver

Reply:
Created a function inside SnakeAndLadderGameService to show leaderboard after game ends